pkg/platform/runtime/artifact/buildPlan.go

```diff
@@ -0,0 +1,215 @@
+package artifact
+
+import (
+	"fmt"
+
+	"github.com/ActiveState/cli/internal/locale"
+	"github.com/ActiveState/cli/internal/logging"
+	"github.com/ActiveState/cli/internal/multilog"
+	model "github.com/ActiveState/cli/pkg/platform/api/graphql/model/buildplanner"
+	monomodel "github.com/ActiveState/cli/pkg/platform/model"
+	"github.com/go-openapi/strfmt"
+	"github.com/thoas/go-funk"
+)
+
+// ArtifactBuildPlan comprises useful information about an artifact that we extracted from a build plan
+type ArtifactBuildPlan struct {
+	ArtifactID       ArtifactID
+	Name             string
+	Namespace        string
+	Version          *string
+	RequestedByOrder bool
+
+	GeneratedBy strfmt.UUID
+
+	Dependencies []ArtifactID
+}
+
+// ArtifactBuildPlanMap maps artifact ids to artifact information extracted from a build plan
+type ArtifactBuildPlanMap map[ArtifactID]ArtifactBuildPlan
+
+// ArtifactNamedBuildPlanMap maps artifact names to artifact information extracted from a build plan
+type ArtifactNamedBuildPlanMap map[string]ArtifactBuildPlan
+
+// NameWithVersion returns a string <name>@<version> if artifact has a version specified, otherwise it returns just the name
+func (a ArtifactBuildPlan) NameWithVersion() string {
+	version := ""
+	if a.Version != nil {
+		version = fmt.Sprintf("@%s", *a.Version)
+	}
+	return a.Name + version
+}
+
+func NewMapFromBuildPlan(build *model.Build) ArtifactBuildPlanMap {
+	if build == nil {
+		return nil
+	}
+
+	res := make(ArtifactBuildPlanMap)
+
+	lookup := make(map[strfmt.UUID]interface{})
+
+	for _, artifact := range build.Artifacts {
+		lookup[artifact.TargetID] = artifact
+	}
+	for _, step := range build.Steps {
+		lookup[step.TargetID] = step
+	}
+	for _, source := range build.Sources {
+		lookup[source.TargetID] = source
+	}
+
+	var terminalTargetIDs []strfmt.UUID
+	for _, terminal := range build.Terminals {
+		terminalTargetIDs = append(terminalTargetIDs, terminal.TargetIDs...)
+	}
+
+	for _, id := range terminalTargetIDs {
+		buildMap(id, lookup, res)
+	}
+
+	return res
+}
+
+func buildMap(baseID strfmt.UUID, lookup map[strfmt.UUID]interface{}, result ArtifactBuildPlanMap) {
+	target := lookup[baseID]
+	artifact, ok := target.(*model.Artifact)
+	if !ok {
+		multilog.Error("Incorrect target type for id %s", baseID)
+		return
+	}
+
+	if artifact.Status == model.ArtifactNotSubmitted {
```

Comment:
What does this mean?

Reply:
From the BuildPlanner documentation: `Artifact has not been submitted to the build backend.`

I should have added a `TODO` here because this was a placeholder for me to do more handling of artifact statuses that aren't `SUCCEEDED`.

Reply:
Followed up with PB and filed a story for us to discuss.